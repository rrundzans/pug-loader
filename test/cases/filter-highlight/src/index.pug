//- test the argument
:highlight(html)
  <h1>Header</h1>

//- test add language in `pre` tag
:highlight
  <pre class="block">
    <code class="language-js">
      const obj = { key: 'value' };
    </code>
  </pre>

//- test override language in `pre` tag from `code`
:highlight
  <pre class="block language-java">
    <code class="language-js">
      const obj = { key: 'value' };
    </code>
  </pre>

//- test language detection from `pre` tag
:highlight
  <pre class="language-js">
    <code>
      const obj = { key: 'value' };
    </code>
  </pre>

//- test language detection from `code` or `pre` tag
:highlight
  <pre class="language-dart">
    <code class="language-dart">
      const obj = { key: 'value' };
    </code>
  </pre>

//- test lazy load of language and dependencies
:highlight
  <pre>
    <code class="language-php">
      $arr = [1, 2];
    </code>
  </pre>

//- test unsupported language `meta`
:highlight
  <pre>
    <code class="language-meta">
      const id = 123;
    </code>
  </pre>