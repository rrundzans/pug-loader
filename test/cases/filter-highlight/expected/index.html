<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>h1</span><span class="token punctuation">></span></span>Header<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>h1</span><span class="token punctuation">></span></span>
<pre class="block language-js">  <code class="language-js">
    <span class="token keyword">const</span> obj <span class="token operator">=</span> <span class="token punctuation">{</span> <span class="token literal-property property">key</span><span class="token operator">:</span> <span class="token string">'value'</span> <span class="token punctuation">}</span><span class="token punctuation">;</span>
  </code>
</pre>
<pre class="block language-js">  <code class="language-js">
    <span class="token keyword">const</span> obj <span class="token operator">=</span> <span class="token punctuation">{</span> <span class="token literal-property property">key</span><span class="token operator">:</span> <span class="token string">'value'</span> <span class="token punctuation">}</span><span class="token punctuation">;</span>
  </code>
</pre>
<pre class="language-js">  <code>
    <span class="token keyword">const</span> obj <span class="token operator">=</span> <span class="token punctuation">{</span> <span class="token literal-property property">key</span><span class="token operator">:</span> <span class="token string">'value'</span> <span class="token punctuation">}</span><span class="token punctuation">;</span>
  </code>
</pre>
<pre class="language-dart">  <code class="language-dart">
    <span class="token keyword">const</span> obj <span class="token operator">=</span> <span class="token punctuation">{</span> key<span class="token punctuation">:</span> <span class="token string-literal"><span class="token string">'value'</span></span> <span class="token punctuation">}</span><span class="token punctuation">;</span>
  </code>
</pre>
<pre class="language-php">  <code class="language-php">
    <span class="token variable">$arr</span> <span class="token operator">=</span> <span class="token punctuation">[</span><span class="token number">1</span><span class="token punctuation">,</span> <span class="token number">2</span><span class="token punctuation">]</span><span class="token punctuation">;</span>
  </code>
</pre>
<pre class="language-meta">  <code class="language-meta">
    const id = 123;
  </code>
</pre>